snuba/datasets/entities/events.py

```diff
@@ -169,6 +169,10 @@ def callback_func(
 
             reason = assign_reason_category(result_data, primary_result_data, referrer)
 
+            # Avoid noise in Sentry
+            if reason == "NONDETERMINISTIC_QUERY":
```

Comment:
Why skip the metric increment? Isn't that still useful?

Reply:
Good point, I didn't mean to do that. Moving this to after the increment.

Reply:
Also added back cache hits here so our metrics still give a complete picture, even though we don't send those to Sentry.